fix: Fix logger, Yarn 2 and components structure

maciejopalinski opened the pull request.

maciejopalinski committed:
fix: Fix custom logger and env vars

maciejopalinski committed:
fix: Fix `Router` component

maciejopalinski committed:
fix: Fix `.lintstagedrc`

maciejopalinski committed:
fix: Fix Yarn 2 & Husky setup

maciejopalinski committed:
refactor: Simplify components structure

Veryyapeee commented on src/utils/Logger.ts:
```diff
@@ -1,22 +1,15 @@
-            return;
-        }
-        console.log(`[${tag}] : ${message}`, { context });
+    static debug(...params: unknown[]) {
```
Don't change that file at all

maciejopalinski replied:
but that's a simpler way of logging, just like console.log works... why complicate stuff that doesn't need to be complicated

Veryyapeee replied:
For standards which is very important for logs, this task has nothing to do with that, recept that changes please

Veryyapeee commented on src/components/Header.tsx:
```diff
@@ -7,10 +7,10 @@ import Toolbar from '@mui/material/Toolbar';
 
 export const Header = () => {
-    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
```
unnecessary changes

maciejopalinski committed:
Merge branch 'main' into fix/logger-yarn-and-structure

maciejopalinski committed:
fix: Remove unnecessary changes

Veryyapeee commented on src/utils/Logger.ts:
```diff
@@ -1,22 +1,16 @@
     }
 
-    static debug(tag: string, message: string, context?: any) {
-        if (process.env.REACT_APP_DEVELOPER_MODE !== 'true') {
```
Why are you removing these ifs

We dont want any logs in production

I dont know why are you even making any changes in this file in scope od this task

maciejopalinski replied:
we don't want errors? yeah ok whatever
and it is not a task, this is a bugfix

maciejopalinski committed:
fix: Disable error logging in production

because why the fuck would we want errors, yeah?
now we can't even reproduce an issue that would potentially happen on our deploy
